lib/index.js

```diff
@@ -1,6 +1,267 @@
-module.exports = exports = require('./promise');
+'use strict';
 
-exports.resolve = require('./resolve');
-exports.reject = require('./reject');
-exports.all = require('./all');
-exports.race = require('./race');
+var immediate = require('immediate');
+
+// istanbul ignore next
+function INTERNAL() {}
+
+const STATE = Symbol('state');
+// 0 is pending
+// 1 is rejected
+// 2 is fullfilled
+
+const QUEUE = Symbol('queue');
+const OUTCOME = Symbol('outcome');
+const HANDLED = Symbol('handled');
+const RESOLVE = Symbol('resolve');
+const REJECT = Symbol('reject');
+const RESOLVE_THENABLE = Symbol('safely resolve thenable');
+const UNWRAP = Symbol('unwrap');
+
+class TryCatch {
```

Comment:
using `class`es is definitely going to balloon the size once/if it's passed through Babel